### PR TITLE
bump k8s.io to 0.28.4

### PR DESCRIPTION
**Pull request by MartinForReal**
bump k8s.io to 0.28.4

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,46 +1,48 @@
 module k8s.io/node-problem-detector
 
-go 1.20
+go 1.21
+
+toolchain go1.21.4
```
Comment (hakman) [inline go.mod:5]: I think this can be removed.

File: test/go.mod [hunk 2]
```diff
@@ -1,64 +1,65 @@
 module k8s.io/node-problem-detector/test
 
-go 1.20
+go 1.21
+
+toolchain go1.21.4
```
Comment (hakman) [inline test/go.mod:5]: I think this can also be removed.

File: go.mod (hunk 1 above)
Comment (MartinForReal) [reply go.mod:5]: k8s.io/test-infra@v0.0.0-20231117021824-70d582860a11 requires go@1.21 Do you think if we should bump to go 1.21?

Comment (hakman) [reply go.mod:5]: That would probably be best.

Comment (MartinForReal) [reply go.mod:5]: I'll keep the change because it is required..

Comment (hakman) [reply go.mod:5]: I am curious, why is it required?

Comment (MartinForReal) [reply go.mod:5]: I'm not sure if it is added by mistake.
https://github.com/kubernetes/test-infra/blob/59f3faaf5f0494159e6c56c04882db53563e2707/go.mod#L213

Comment (hakman) [reply go.mod:5]: we need to bump to 1.21 for sure, just not have the toolchain directive (I think I was not very clear).

Comment (MartinForReal) [reply go.mod:5]: Oh. I see.

Comment (MartinForReal) [reply go.mod]: Thanks for the review! It has been removed.

File: test/go.mod (hunk 2 above)
Comment (MartinForReal) [reply test/go.mod]: it has been removed.

**Comment by hakman** [comment]
One more thing @MartinForReal, I don't think we want to add the vendor folder for the test dir. Could you remove?

**Comment by MartinForReal** [comment]
hi @vteratipally Do you have any advice on this? Thanks!

**Comment by vteratipally** [comment]
updated to v0.29.0
/close

**Pull request closed without merging**